gef.py

```diff
@@ -806,6 +867,28 @@ def heap_addr(self):
         _addr = int(self) + self.struct_size
         return malloc_align_address(_addr)
 
+    def get_heap_infos(self):
+        if self.is_main_arena():
+            return None
+        heap_addr = self.get_heap_for_ptr(self.top)
+        heap_infos = [GlibcHeapInfo(heap_addr)]
+        while heap_infos[-1].prev != 0:
+            prev = int(heap_infos[-1].prev)
+            heap_info = GlibcHeapInfo(prev)
+            heap_infos.append(heap_info)
+        return heap_infos[::-1]
+
+    @staticmethod
+    def get_heap_for_ptr(ptr):
+        """Find the corresponding heap for a given pointer (int).
+        See https://github.com/bminor/glibc/blob/glibc-2.34/malloc/arena.c#L129"""
+        if is_32bit():
+            default_mmap_threshold_max = 512 * 1024
+        elif is_64bit():
```

Comment:
what if it's neither? Can it happen?

Reply:
hm, not sure. In the sources for glibc it's checking if its 32bit or else (so not explicitly checking for 64bit). If you want I can adopt that behavior.